docs/smart-contracts/modules.md

```diff
@@ -0,0 +1,564 @@
+
+Modules can implement interfaces that include the following components:
+
+- [function](/reference/syntax#defun) signatures
+- [constant](/reference/syntax#defconst) values
+- [schema](/reference/syntax#defschema) definitions
+- [defpact](/reference/syntax#defpact) specifications
+- [models](/reference/property-checking) for properties
+- [capability](/build/pact/advanced#capabilities) specifications
+- [imported definitions](/reference/syntax#use) from other modules
+
+The following example illustrates how to declare and implement an interface with one function signature and one constant value:
+
+```pact
+(interface my-interface
+    (defun hello-number:string (number:integer)
+      @doc "Return the string \"Hello, $number!\" when given a string"
+        )
+
+    (defconst SOME_CONSTANT 3)
+)
+
+(module my-module (read-keyset 'my-keyset)
+    (implements my-interface)
+
+    (defun hello-number:string (number:integer)
+        (format "Hello, {}!" [number]))
+
+    (defun square-three ()
+        (* my-interface.SOME_CONSTANT my-interface.SOME_CONSTANT))
+)
+```
+
+As you can see in this example, the `my-module` module implements the `hello-number` function signature. 
+The constant declared in the interface is accessed directly by its fully qualified name `namespace.interface.const`.
+
+## Implements and use keywords
+
+You can create complex and layered relationships between interfaces and modules.
+One important way you can create this layered relationships is by using combinations of the `implements` and `use` keywords.
+The `use` keyword enables you to import elements from the specified `module` into a namespace, interface, or module. 
+For example, you can specify the `use` keyword in an interface declaration to import table schemas and types from a specified module. 
+
+You can also include `use` statements at the top-level of a contract or within a module declaration to make all or parts of a specified module available in the current module context. 
+For example, you can specify a list of functions, constants, and schema names to import from the specified `module`. 
+If you explicitly define the function, constant, and schema names to import, only those items are available in the module body. 
+
+You can also specify a `hash` argument in `use` statements to check that an imported module's hash matches the `hash` you expect, and fail if the hashes are not the same. 
+By including the `hash` argument in a `use` statement, you can perform a simplified form of version control or dependency checking.
+
+
+The following example is an excerpt from the `marmalade-v2.ledger` module that illustrates the relationships created by combining `implements` and `use` statements.
+In this example, `marmalade-v2` is the primary namespace where the `ledger` contract is deployed.
+
+```pact
+(namespace (read-string 'ns))
+
+(module ledger GOVERNANCE
+
+  @model
```

Comment:
I think we should remove all @model annotations as we are not maintaining the FV system

Reply:
I lifted this directly from marmalade code and wasn't sure whether to remove the @model stuff since there's a lot of them in Marmalade and SpireKey code.